### ISSUE TITLE
ProgressBar IsIndeterminate memory leak

### ISSUE THREAD

**Issue by alovegit** [AvaloniaUI/Avalonia#12922]
**Describe the bug**
Having a ProgressBar with IsIndeterminate="True" causes it and the visual tree to be rooted in animation.

**To Reproduce**

1. Have a UserControl that contains a ProgressBar with IsIndeterminate="True".
2. On the main view, have a ContentControl and set its Content property to an instance of the UserControl from step 1.
3. From code, set the ContentControl's Content property to null.
4. The ProgressBar is now rooted.

**Expected behavior**
The ProgressBar is no longer rooted and can be garbage collected.

My workaround was to associate a Behavior on the ProgressBar that sets IsIndeterminate to false when OnDetachedFromVisualTree occurs.

**Desktop (please complete the following information):**

- OS: Windows
- Version 11.0.4

**Comment by timunie** [comment]
Duplicate of #2881

**Comment by alovegit** [comment]
@timunie I don't see how this is the same issue. The other ticket is referring to an animation of the Value property causing memory to grow. My issue is that the tree gets rooted even when detached from the tree, which is different.

**Comment by timunie** [comment]
I thought it's the same root cause -> Animation is leaking. But I can reopen if you feel it is a different root cause. @jmacato may know more

**Comment by alovegit** [comment]
Yes, definitely different in my opinion.  I don't think this is a ProgressBar only issue either, as all animations should be torn down once its associated control is removed from the visual tree, and then recreated again if the control gets re-added to the visual tree.

**Comment by timunie** [comment]
exactly. general animation issue.
